u5-rest-apis/feature/covid-controller.js

```diff
@@ -0,0 +1,123 @@
+const { db } = require('../_services/firebase-admin-initialized')
+const { firestore } = require('firebase-admin')
+
+const readCovidR = async (req, res) => {
+  try {
+    // 1. Inputs
+    // none
+
+    // 2. Query
+    const query = db.collection('covid-latest').get()
+
+    // 3. Response
+    const payload = (await query)
+      .docs.map(doc => doc.data())
+      .map(data => ({ date: data.date.toMillis(), stateId: data.stateId, stateName: data.stateName, cases: data.cases, casesNew: data.casesNew, vaccineOne: data.vaccineOne, vaccineOnePercent: data.vaccineOnePercent, vaccineComplete: data.vaccineComplete, vaccineCompletePercent: data.vaccineCompletePercent }))
+
+    res.json({
+      result: 'ok',
+      payload: payload,
+      count: payload.length
+    })
+  } catch (err) {
+    console.error(err)
+    res.status(500).json({
+      result: 'error',
+      payload: [],
+      count: 0
+    })
+  }
+}
+
+const readCovid = async (req, res) => {
+  try {
+    // 1. Inputs
+    const stateId = req.params.stateId.toUpperCase()
+
+    // 2. Query
+    const query = db.collection('covid-history').doc(stateId).get()
+
+    // 3. Response
+    const snapshot = await query
+    if (!snapshot.exists) { return res.status(404).json({ result: 'not found' }) }
+
+    const { stateName, history } = snapshot.data()
+    const payload = { stateId, stateName, history: history.map(item => ({ date: item.date.toMillis(), cases: item.cases, casesNew: item.casesNew, vaccineOne: item.vaccineOne, vaccineOnePercent: item.vaccineOnePercent, vaccineComplete: item.vaccineComplete, vaccineCompletePercent: item.vaccineCompletePercent })) }
+
+    res.json({ result: 'ok', payload })
+  } catch (err) {
+    console.error(err)
+    res.status(500).json({ result: 'error' })
+  }
+}
+
+const createCovid = async (req, res) => {
+  try {
+    // 1. Inputs
+    const { stateId, date, cases, casesNew, vaccineOne, vaccineOnePercent, vaccineComplete, vaccineCompletePercent } = req.body
+    const record = {
+      stateId,
+      date: firestore.Timestamp.fromMillis(date),
```

Comment:
ฉันพบวิธีที่ชาญฉลาดในการจัดการการเปลี่ยนแปลงวันที่หลังจากที่เราได้พูดคุยกัน ฉันชอบเพราะชื่อตัวแปรมีความชัดเจนมากกว่า และฉันรู้สึกสบายใจที่จะเขียนบรรทัดสุดท้ายในบรรทัดเดียว:
I found a clever way to manage the date transformation after we spoke. I prefer it because the variable names are clearer, and I feel more comfortable writing the last line on a single line:
```js
const { stateId, date: dateMillis, cases, ... } = req.body // เปลี่ยนชื่อ date เป็น dateMillis
const date = firestore.Timestamp.fromMillis(dateMillis)
const record = { stateId, date, cases, ... } // ทั้งหมดเป็นแบบสั้น
```